### PR TITLE
dpl: check if db_inst_ exists before accessing it

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/5205

### CODE REVIEW

**Comment by eder-matheus** [comment]
@maliberty I found that the db_inst being nullptr was the cause of the issue, but I couldn't see why it is nullptr at this point. The position of the Cell that is returned by the pixel doesn't have any cell on it.

**Review by maliberty — APPROVED** [review]
Let's take this as a quick fix and dig in more afterwards

**Comment by maliberty** [comment]
Which issue is this related to?

**Comment by eder-matheus** [comment]
> Which issue is this related to?

This issue: https://github.com/The-OpenROAD-Project/OpenROAD/issues/5205
The only difference I first noticed is using REGION and GROUP, but we don't have this specific crash in other test cases from the same user.

**Comment by maliberty** [comment]
The cell is the dummy_cell used by groupInitPixels and groupInitPixels2.  It is used to block the area outside the region during placement of the group.  I think it is better to move the check into isBlock itself.  

isBlock should be a method of Cell not Opendp.

File: src/dpl/src/Place.cpp [hunk 1]
```diff
@@ -1364,7 +1364,7 @@ DbuPt Opendp::legalPt(const Cell* cell, const bool padded) const
     // If that didn't do the job fall back on the old move to nearest
     // edge strategy.  This doesn't consider site availability at the
     // end used so it is secondary.
-    if (block && isBlock(block)) {
+    if (block && block->db_inst_ && isBlock(block)) {
```
Comment (maliberty) [inline src/dpl/src/Place.cpp:1367]: Move the check to isBlock

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/dpl/src/Place.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/dpl/src/Place.cpp]: Done, both moving the check to isBlock and move isBlock to Cell struct.

**Review by maliberty — APPROVED** [review]

**Pull request merged**